src/System.Runtime/tests/System/Guid.cs

```diff
@@ -208,4 +115,311 @@ public static void TestRandomness()
         // anything special for it.
         RandomDataGenerator.VerifyRandomDistribution(random);
     }
+
+    [Theory]
+    [MemberData(nameof(GuidStrings_Valid_TestData))]
+    public static void TestParse(string input, string format, Guid expected)
+    {
+        Assert.Equal(expected, Guid.Parse(input));
+        Assert.Equal(expected, Guid.ParseExact(input, format.ToUpperInvariant()));
+        Assert.Equal(expected, Guid.ParseExact(input, format.ToLowerInvariant())); // Format should be case insensitive
+
+        Guid result1;
+        Assert.True(Guid.TryParse(input, out result1));
+        Assert.Equal(expected, result1);
+
+        Guid result2;
+        Assert.True(Guid.TryParseExact(input, format.ToUpperInvariant(), out result2));
+        Assert.Equal(expected, result2);
+
+        Guid result3;
+        Assert.True(Guid.TryParseExact(input, format.ToLowerInvariant(), out result3)); // Format should be case insensitive
+        Assert.Equal(expected, result3);
+    }
+
+    [Theory]
+    [MemberData(nameof(GuidStrings_Invalid_TestData))]
+    public static void TestParse_Invalid(string input, Type exceptionType)
+    {
+        // Overflow exceptions throw as format exceptions in Parse
+        if (exceptionType.Equals(typeof(OverflowException)))
+        {
+            exceptionType = typeof(FormatException);
+        }
+        Assert.Throws(exceptionType, () => Guid.Parse(input));
+        Assert.Throws(exceptionType, () => Guid.ParseExact(input, "N"));
+        Assert.Throws(exceptionType, () => Guid.ParseExact(input, "D"));
+        Assert.Throws(exceptionType, () => Guid.ParseExact(input, "B"));
+        Assert.Throws(exceptionType, () => Guid.ParseExact(input, "P"));
+        Assert.Throws(exceptionType, () => Guid.ParseExact(input, "X"));
+
+        // Sometimes Guid.TryParse actually DOES throw - if this happens we can ignore this
+        Guid result;
+        try
+        {
+            Assert.False(Guid.TryParse(input, out result));
+        }
+        catch (FormatException)
+        {
+        }
+
+        try
+        {
+            Assert.False(Guid.TryParseExact(input, "N", out result));
+            Assert.False(Guid.TryParseExact(input, "D", out result));
+            Assert.False(Guid.TryParseExact(input, "B", out result));
+            Assert.False(Guid.TryParseExact(input, "P", out result));
+            Assert.False(Guid.TryParseExact(input, "X", out result));
+        }
+        catch (FormatException)
```

Comment:
`TryParse("{0xdddddddd, 0xdddd, 0xdddd,{0xdd0xdd,0xdd,0xdd,0xdd,0xdd,0xdd,0xdd}}");` is an example